api/workflow_create.go

```diff
@@ -0,0 +1,40 @@
+package api
+
+import (
+	"github.com/mesg-foundation/core/systemservices/workflow"
+	"github.com/mesg-foundation/core/utils/workflowparser/yaml"
+)
+
+// CreateWorkflow creates and runs a new workflow with optionally given unique name.
+func (a *API) CreateWorkflow(definition yaml.WorkflowDefinition, name string) (id string, err error) {
+	return newWorkflowCreator(a).Create(definition, name)
+}
+
+// workflowCreator provides functionalities to create and run a new workflow.
+type workflowCreator struct {
+	api *API
+}
+
+// newWorkflowCreator returns a new workflow creator with given api.
+func newWorkflowCreator(api *API) *workflowCreator {
+	return &workflowCreator{
+		api: api,
+	}
+}
+
+// Create creates and runs a new workflow with optionally given unique name.
+func (w *workflowCreator) Create(definition yaml.WorkflowDefinition, name string) (id string, err error) {
```

Comment:
Move Create method body to CreateWorkflow

Reply:
I kind of agree with this, but we follow the same pattern all around the _api_. I think, for now we should keep this one as it's and if we want to make a change about this pattern we can do it for all methods together in a different PR.

Reply:
There is no need to keep it and then change all at once, we can start one by one and get rid of it eventually.

Reply:
I don't think it's something that we need to get rid of. To me, having it good to isolate funcs from different groups, otherwise they all will be defined globally which kills the isolation, causes poor naming on funcs and opens up the possibility of overlapping func names. If we can find a better way, we can switch to it but by having a discussion first. We can think about a guideline and maybe open a forum proposal.

Reply:
I agree with @ilgooz on this one